### PR TITLE
feat(appeals): DB refactor

### PR DESCRIPTION
**Pull request by fabfab-kc**
## Describe your changes
This is a major DB refactoring, in preparation for integration tasks, with 3 goals:

1. Aligning DB structure to the data models
2. Improving TS definitions and minimise warnings
3. Adjust AV scan following the document upload re-work

Additionally, implements adding the correct metadata on file upload.

## Issue ticket number and link

Integration tickets (being moved around)

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/web/src/server/appeals/appeal-details/costs/__tests__/__snapshots__/costs.test.js.snap [hunk 1]
```diff
@@ -1151,6 +1177,19 @@ exports[`costs GET /costs/:costsCategory/manage-documents/:folderId/:documentId
 
 exports[`costs GET /costs/:costsCategory/manage-documents/:folderId/:documentId should render the manage individual document page with the expected content if the folderId and documentId are both valid and the document virus check status is "not_checked" (lpa) 1`] = `
```
[the remark this answers is not in the document]
Comment (fabfab-kc) [reply appeals/web/src/server/appeals/appeal-details/costs/__tests__/__snapshots__/costs.test.js.snap:1178]: Yes I have, they now all come from here
image

File: appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js [hunk 2]
```diff
@@ -99,53 +100,56 @@ const updateAppellantCaseValidationOutcome = async (
...
-			await notifyClient.sendEmail(config.govNotify.template.appealIncomplete, recipientEmail, {
-				appeal_reference_number: appeal.reference,
-				site_address: siteAddress,
-				due_date: formatDate(appealDueDateAsDateObject, false),
-				reasons: incompleteReasonsList
-			});
-		} catch (error) {
-			if (error) {
-				throw new Error(ERROR_FAILED_TO_SEND_NOTIFICATION_EMAIL);
+		if (isOutcomeIncomplete(validationOutcome.name) && updatedAppeal) {
```
Comment (BiggsKC) [inline appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js:106]: `&& updatedAppeal` can be removed

File: appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js [hunk 3]
```diff
@@ -99,53 +100,56 @@ const updateAppellantCaseValidationOutcome = async (
...
+						throw new Error(ERROR_FAILED_TO_SEND_NOTIFICATION_EMAIL);
+					}
+				}
+			}
 		}
 
-		const invalidReasonsList = getFormattedReasons(
-			updatedAppellantCase?.appellantCaseInvalidReasonsOnAppellantCases
-		);
+		if (isOutcomeInvalid(validationOutcome.name) && updatedAppeal) {
```
Comment (BiggsKC) [inline appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js:133]: `&& updatedAppeal` can be removed

**Review by glpins — DISMISSED** [review]

**Review by BiggsKC — DISMISSED** [review]
All good

**Review by kandcdev — APPROVED** [review]

**Review by BiggsKC — APPROVED** [review]
All good

**Pull request merged**